### ISSUE TITLE
Upgrade to Laravel 7

### ISSUE THREAD

**Issue by Sharvadze** [cretueusebiu/laravel-nuxt#103]
Would be useful to upgrade the package to Laravel 7

**Comment by vahidalvandi** [comment]
laravel 7 is not LTS

**Comment by zlodes** [comment]
@vahidalvandi but who cares? :)

**Comment by ZapevalovAnton** [comment]
Laravel 7 introduced native support for CORS. "spatie/laravel-cors" package is not needed. It is necessary to rewrite the part where CORS is used. And remove this package from dependencies.

**Comment by ZapevalovAnton** [comment]
@cretueusebiu any ideas?

**Comment by JayCasee** [comment]
I've finished to update Laravel 7 as below procedures.
**But I'm not sure whether it was a right approach or not.**
So if anyone find any issues, please share for preventing other persons to spend time for this.

**1. Update composer.json.**

In my case, I've update as below.
There are something that I personally need (e.g. nuwave/lighthouse), so please be aware of it.
```
"require": {
  "php": "^7.2.5",
  "fideloper/proxy": "^4.2",
  "laravel/socialite": "^4.0",
  "tymon/jwt-auth": "^1.0.0-rc.5",
  "fruitcake/laravel-cors": "^1.0",
  "guzzlehttp/guzzle": "^6.3",
  "laravel/framework": "^7.0",
  "laravel/tinker": "^2.0",
  "nuwave/lighthouse": "^4.12"
},
"require-dev": {
  "beyondcode/laravel-dump-server": "^1.0",
  "filp/whoops": "^2.0",
  "facade/ignition": "^2.0",
  "fzaninotto/faker": "^1.9.1",
  "laracademy/generators": "^2.3",
  "laravel/ui": "^2.0",
  "mockery/mockery": "^1.3.1",
  "nunomaduro/collision": "^4.1",
  "phpunit/phpunit": "^8.5"
},
```

Do not update Composer yet, you need the following issues, before the Composer update.

**2. Solve Symfony 5 related upgrades.**

I've just followed this link. >>> https://stackoverflow.com/questions/50840960/script-php-artisan-packagediscover-handling-the-post-autoload-dump-event-retur >>> Answered by Oginga Michael
The core things that you need to do is...

In App\Exceptions\Handler:
```
//Use Exception;
Use Throwable;
```

Then methods to accept instances of Throwable instead of Exceptions as follows:
```
//public function report(Exception$exception);
public function report(Throwable $exception); # When you do this, remove the semicolon.
//public function render($request, Exception $exception);
public function render($request, Throwable $exception); # When you do this, remove the semicolon.
```

In config\session.php:
```
//'secure' => env('SESSION_SECURE_COOKIE', false),
'secure' => env('SESSION_SECURE_COOKIE', null),
```

Then, run the Composer update.

3. Change CORS from Spatie to Fruitcake

In app/Http/Kernel.php:
```
# \Spatie\Cors\Cors::class,
\Fruitcake\Cors\HandleCors::class,
```

Also, if you need, please set config/cors.php as your needs.


FY, I've done this under a super sleepy status, so if I missed some part, please let me know.

**Comment by cretueusebiu** [comment]
Fixed in [3.0.0](https://github.com/cretueusebiu/laravel-nuxt/releases/tag/3.0.0)
